api/src/main/java/org/pmiops/workbench/rdr/RdrExportServiceImpl.java

```diff
@@ -0,0 +1,263 @@
+              "Error while trying to fetch modified/created user list: %s", ex.getMessage()));
+    }
+    return userIdList;
+  }
+
+  /**
+   * Retrieve the list of all workspace ids that are either a) not in rdr_Export table or b) have
+   * last_modified_time (workspace table) > export_time (rdr_export table)
+   *
+   * @return list of Workspace Ids
+   */
+  @Override
+  public List<Long> findAllWorkspacesIdsToExport() {
+    List<Long> workspaceListToExport = new ArrayList<Long>();
+    try {
+      workspaceListToExport =
+          rdrExportDao.findDbWorkspaceIdsToExport().stream()
+              .map(workspaceId -> workspaceId.longValue())
+              .collect(Collectors.toList());
+    } catch (Exception ex) {
+      log.severe(
+          String.format(
+              "Error while trying to fetch modified/created workspace list: %s", ex.getMessage()));
+    }
+    return workspaceListToExport;
+  }
+
+  /**
+   * Call the Rdr API to send researcher data and if successful store all the ids in rdr_export
+   * table with current date as the lastExport date
+   *
+   * @param userIds
+   */
+  @Override
+  public void exportUsers(List<Long> userIds) {
+    List<RdrResearcher> RdrResearchersList;
+    try {
+      RdrResearchersList =
+          userIds.stream()
+              .map(userId -> toRdrResearcher(userDao.findUserByUserId(userId)))
+              .collect(Collectors.toList());
+      // TODO: REMOVE DEBUGGER
+      RdrApiProvider.get().getApiClient().setDebugging(true);
+      RdrApiProvider.get().exportResearcheres(RdrResearchersList);
+
+      updateDBRdrExport(RdrEntity.USER, userIds);
+    } catch (ApiException ex) {
+      log.severe("Error while sending researcher data to RDR");
+    }
+  }
+
+  /**
+   * Call the Rdr API to send researcher data and if successful store all the ids in rdr_export
+   * table with current date as the lastExport date
+   *
+   * @param workspaceIds
+   */
+  @Override
+  public void exportWorkspaces(List<Long> workspaceIds) {
+    List<RdrWorkspace> RdrWorkspacesList;
+    try {
+      RdrWorkspacesList =
+          workspaceIds.stream()
+              .map(
+                  workspaceId ->
+                      toRdrWorkspace(workspaceDao.findDbWorkspaceByWorkspaceId(workspaceId)))
+              .collect(Collectors.toList());
+      RdrApiProvider.get().getApiClient().setDebugging(true);
+      RdrApiProvider.get().exportWorkspaces(RdrWorkspacesList);
+      updateDBRdrExport(RdrEntity.WORKSPACE, workspaceIds);
+    } catch (ApiException ex) {
+      log.severe("Error while sending workspace data to RDR");
+    }
+  }
+
+  // Convert workbench DBUser to RDR Model
+  private RdrResearcher toRdrResearcher(DbUser dbUser) {
+    RdrResearcher researcher = new RdrResearcher();
+    researcher.setUserId((int) dbUser.getUserId());
+    researcher.setCreationTime(dbUser.getCreationTime().toLocalDateTime().atOffset(offset));
+    if (dbUser.getLastModifiedTime() != null)
```

Comment:
Is there a reason why we check for null values for lastModifiedTime but not for creationTime? It seems slightly inconsistent that we don't handle these timestamp fields in the same way.